dspace-spring-rest/src/test/java/org/dspace/app/rest/EPersonRestRepositoryIT.java

```diff
@@ -514,4 +520,480 @@ public void deleteViolatingConstraints() throws Exception {
+
+        getClient(token).perform(get("/api/eperson/epersons/" + ePerson.getID()))
+                        .andExpect(status().isOk())
+                        .andExpect(jsonPath("$.email", Matchers.is("johndoe@fake-email.com")))
+                        .andExpect(jsonPath("$.netid", Matchers.nullValue()));
+
+    }
+
+    @Test
+    public void patchNetIdMissingValue() throws Exception {
+
+        context.turnOffAuthorisationSystem();
+
+        EPerson ePerson = EPersonBuilder.createEPerson(context)
+                                        .withNameInMetadata("John", "Doe")
+                                        .withEmail("Johndoe@fake-email.com")
+                                        .withNetId("testId")
+                                        .build();
+
+        String newId = "newId";
+
+        String token = getAuthToken(admin.getEmail(), password);
+
+        List<Operation> ops = new ArrayList<Operation>();
+        ReplaceOperation replaceOperation = new ReplaceOperation("/netid", newId);
+        ops.add(replaceOperation);
+        String patchBody = getPatchContent(ops);
+
+        // initialize to true
+        getClient(token).perform(patch("/api/eperson/epersons/" + ePerson.getID())
+                .content(patchBody)
+                .contentType(MediaType.APPLICATION_JSON_PATCH_JSON))
+                        .andExpect(status().isOk())
+                        .andExpect(jsonPath("$.netid", Matchers.is(newId)));
+
+
+        List<Operation> ops2 = new ArrayList<Operation>();
+        ReplaceOperation replaceOperation2 = new ReplaceOperation("/netid", null);
+        ops2.add(replaceOperation2);
+        patchBody = getPatchContent(ops2);
+
+        // should return bad request
+        getClient(token).perform(patch("/api/eperson/epersons/" + ePerson.getID())
+                .content(patchBody)
+                .contentType(MediaType.APPLICATION_JSON_PATCH_JSON))
+                        .andExpect(status().isBadRequest());
+
+        // value should be unchanged.
+        getClient(token).perform(get("/api/eperson/epersons/" + ePerson.getID()))
+                        .andExpect(status().isOk())
+                        .andExpect(jsonPath("$.netid", Matchers.is(newId)))
+                        .andExpect(jsonPath("$.email", Matchers.is("johndoe@fake-email.com")))
+                        .andExpect(jsonPath("$.canLogIn", Matchers.is(false)));
+
+
+    }
+
+    @Test
+    public void patchCanLogin() throws Exception {
+
+        context.turnOffAuthorisationSystem();
+
+        EPerson ePerson = EPersonBuilder.createEPerson(context)
+                                        .withNameInMetadata("John", "Doe")
+                                        .withEmail("Johndoe@fake-email.com")
+                                        .build();
+
+        List<Operation> ops = new ArrayList<Operation>();
+        ReplaceOperation replaceOperation = new ReplaceOperation("/canLogin", "true");
```

Comment:
the right way to patch a boolean attribute is to send a real boolean instead than a String.
I suggest to make the implementation client friendly here and accept both literal boolean than "textual" boolean.
See here for an example of such implementation in the submission patches
https://github.com/DSpace/DSpace/commit/89ebd2c154515b8bc41ca6fbab199dfb097399e8

You could also add an additional test to check that textual boolean are still accepted and change this test to check the right boolean patch